HealthcareApp/src/Healthcare.Application/Appointments/Commands/Book/BookAppointmentCommandHandler.cs

```diff
@@ -0,0 +1,60 @@
+using Application.Abstractions;
+using Application.Abstractions.Decorators;
+using Domain.Entities;
+using Healthcare.Application.Appointments.Commands.Book;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using static Domain.Constants.AppointmentStatusConstants;
+
+namespace Healthcare.Application.Appointments.Commands;
+
+public class BookAppointmentCommandHandler : IRequestHandler<BookAppointmentCommand, bool>
+{
+    private readonly IUserManagerDecorator<ApplicationUser> _userManager;
+    private readonly IAppointmentRepository _appointmentRepository;
+
+
+    public BookAppointmentCommandHandler(IUserManagerDecorator<ApplicationUser> userManager,
+        IAppointmentRepository appointmentRepository)
+    {
+        _userManager = userManager;
+        _appointmentRepository = appointmentRepository;
+    }
+
+    public async Task<bool> Handle(BookAppointmentCommand request, CancellationToken cancellationToken)
+    {
+        var users = await _userManager.GetAllUsersAsync();
+
+        var doctor = await users.FirstOrDefaultAsync(x => x.Id == request.DoctorId, cancellationToken);
+        if (doctor == null) throw new Exception("Doctor not found");
+
+        var patient = await users.FirstOrDefaultAsync(x => x.Id == request.DoctorId, cancellationToken);
+        if (patient == null) throw new Exception("Patient not found");
+
+        var endTime = request.StartTime.Add(TimeSpan.FromMinutes(15));
+
+
+        var appointments = await _appointmentRepository.GetAllAppointmentsAsync();
```

Comment:
The same problem as with users - don't ever get all your database records in memory, this is terrible idea